src/sys/socket/mod.rs

```diff
@@ -392,11 +392,15 @@ impl Ipv6MembershipRequest {
 /// Create a buffer large enough for storing some control messages as returned
 /// by [`recvmsg`](fn.recvmsg.html).
 ///
+/// The usage of space created by `cmsg_space!` depends on specific `capacity` set on the `Vec`
```

Comment:
I think the main point of this PR is that `cmsg_space` leaks the implementation detail into the API, which user needs to be warned about.

Reply:
That's a fair point.  Instead of just tweaking the documentation, can you think of a way to fix `cmsg_space!` to this problem can't recur?  I think it shouldn't be hard; make `cmsg_space!` return a data type that implements `Deref<Vec<u8>>` but isn't `Clone`.  It would even be backwards compatible, since `recvmsg` is already generic.